### ISSUE TITLE
Inconsistency between `.find.watch` and `.find.fetch`

### ISSUE THREAD

**Issue by mlucy** [rethinkdb/horizon#622]
When I write this:

``` js
hz('test').find(0).watch().subscribe(msg => console.log(msg))
```

It prints `null`.

But when I write this:

``` js
hz('test').find(0).fetch().subscribe(msg => console.log(msg))
```

It prints nothing.  If I use `findAll` instead both `watch` and `fetch` print `[]`.  Ping @Tryneus because he said it might be a bug.

**Comment by deontologician** [comment]
@mlucy Can you capture the protocol stuff for these two commands? I would tentatively guess either we aren't receiving a complete, or we're mishandling it in the client

**Comment by mlucy** [comment]
@deontologician --

```
4{"type":"subscribe","options":{"collection":"test","find":{"id":0}},"request_id":2}
4{"state":"synced","request_id":2}
4{"type":"query","options":{"collection":"test","find":{"id":0}},"request_id":3}
4{"data":[],"state":"complete","request_id":3}
4{"request_id":3,"type":"end_subscription"}
```

(The first is for `watch`, the second is for `fetch`.)

**Comment by deontologician** [comment]
Ok, looks like a client error as suspected

**Comment by marshall007** [comment]
The issue is with this line in [`client/src/index.js#L121`](https://github.com/rethinkdb/horizon/blob/next/client/src/index.js#L121) because `Observable.from([])` isn't going to emit any values. Shouldn't we expect `{ "data": [null], ... }` in the response for a `find` with no results?

**Comment by marshall007** [comment]
Looks like server behavior was changed here: https://github.com/rethinkdb/horizon/commit/a2dcd5fbf9f3ebd7fbfe3c93f33b04420a3c79a2#diff-860203edebb57e95b7530efa657e58e6L71

**Comment by deontologician** [comment]
Awesome, thanks @marshall007

**Comment by deontologician** [comment]
Fix up for this in #652
